OpenSim/Common/C3DFileAdapter.cpp

```diff
@@ -108,13 +112,18 @@ C3DFileAdapter::extendRead(const std::string& fileName) const {
 
         double time_step{1.0 / acquisition->GetPointFrequency()};
         for(int f = 0; f < marker_nrow; ++f) {
-            SimTK::RowVector_<SimTK::Vec3> row{marker_pts->GetItemNumber()};
+            SimTK::RowVector_<SimTK::Vec3> row{ marker_pts->GetItemNumber(), 
+                                                SimTK::Vec3(SimTK::NaN) };
             int m{0};
             for(auto it = marker_pts->Begin();  it != marker_pts->End(); ++it) {
                 auto pt = *it;
-                row[m++] = SimTK::Vec3{pt->GetValues().coeff(f, 0),
-                                       pt->GetValues().coeff(f, 1),
-                                       pt->GetValues().coeff(f, 2)};
+                if (!( pt->GetValues().row(f).isZero() ||  //not precisely zero
+                    (pt->GetResiduals().coeff(f) == -1))) {//nor is residual -1
```

Comment:
This condition is confusing, which is probably why you wrote the comment. I suggest removing the comment and replacing `if (!( a || b ))` with `if (!a && !b)` which is equivalent and self-commenting. Might be useful to include a comment that explains what's going on here. What does a residual of -1 mean? Perhaps comment before L115: "Initialize table with SimTK::NaN and fill in only the quantities that are valid."